mypyc/ops_str.py

```diff
@@ -31,6 +32,27 @@
     error_kind=ERR_MAGIC,
     emit=simple_emit('{dest} = PyUnicode_Join({args[0]}, {args[1]});'))
 
+
+def emit_str_split(emitter: EmitterInterface, args: List[str], dest: str) -> None:
+    temp = emitter.temp_name()
+    emitter.emit_declaration('Py_ssize_t %s;' % temp)
+    emitter.emit_lines(
+        '%s = CPyTagged_AsSsize_t(%s);' % (temp, args[2]),
+        '%s = PyUnicode_Split(%s, %s, %s);' % (dest, args[0], args[1], temp))
+
+
+str_split_types = [str_rprimitive, str_rprimitive, int_rprimitive]  # type: List[RType]
+str_split_emits = [simple_emit('{dest} = PyUnicode_Split({args[0]}, NULL, -1);'),
+                   simple_emit('{dest} = PyUnicode_Split({args[0]}, {args[1]}, -1);'),
+                   emit_str_split]  # type: List[EmitCallback]
+for i in range(3):
+    method_op(
+        name='split',
+        arg_types=str_split_types[0:i+1],
+        result_type=object_rprimitive,
```

Comment:
Making this `list_rprimitive` will let us generate slightly better code. Otherwise we'll emit a cast on the result to make sure it is actually a list.